### PR TITLE
Add group create service

### LINKED ISSUE #24815: Camunda services support creating a group
### Description
Group creation should be supported in the `camunda-service` module. An example of the implementation of creation via the service module can be found [here](https://github.com/camunda/camunda/pull/20705/commits/7c9c1daddc88155e7e7eec15ca762bc969c41152#diff-3e9df8ac9a5cc69340bd6f5aaf820a98799ee2fa656f69e3cb0cf91fb9dd1931R37)

### Acceptance Criteria
- [x] A `GroupServices` exists in `camunda-services` module extending the `ApiServices` class (Note: this will change when searching is implemented to extending `SearchQueryService`) hint: [UserServices](https://github.com/camunda/camunda/blob/main/service/src/main/java/io/camunda/service/UserServices.java) 
- [x] A `GroupDTO` record exists in the `GroupServices` module containing the fields of the `GroupRecord`
- [x] A `BrokerGroupCreateRequest` exists hint: [BrokerUserCreateRequest](https://github.com/camunda/camunda/blob/main/zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerUserCreateRequest.java)
- [x] A method exists with signature `createGroup(final GroupDTO GroupDTO)`
- [x] A `UserServicesTest` class exists containing unit tests for the `createGroup` method

### LINKED ISSUE #24882: Camunda services support retrieving group(s)

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add a group service together with `create`, `find`, and `get` methods.

ℹ️ The PR also implements the `GroupSearchClient`, as it was required to implement the `GroupService` class fully. As a result, a `search` method is also provided. The `search` method is then used in the `get` and `find` methods. If necessary, I can split these changes into two PRs.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24815
closes #24882

### CODE REVIEW

File: service/src/test/java/io/camunda/service/GroupServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    stubbedBrokerClient = new StubbedBrokerClient();
+    client = mock(GroupSearchClient.class);
+    when(client.withSecurityContext(any())).thenReturn(client);
+    services =
+        new GroupServices(
+            stubbedBrokerClient, mock(SecurityContextProvider.class), client, authentication);
+  }
+
+  @Test
+  public void shouldEmptyQueryReturnRoles() {
```
Comment (nicpuppa) [inline service/src/test/java/io/camunda/service/GroupServiceTest.java:47]: 🔧 `shouldEmptyQueryReturnGroupes`

File: service/src/test/java/io/camunda/service/GroupServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import io.camunda.search.query.SearchQueryResult;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.security.SecurityContextProvider;
+import io.camunda.zeebe.gateway.api.util.StubbedBrokerClient;
+import java.util.List;
+import org.assertj.core.util.Arrays;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class GroupServiceTest {
```
Comment (nicpuppa) [inline service/src/test/java/io/camunda/service/GroupServiceTest.java:28]: ❌ You missed to test the group creation

File: service/src/test/java/io/camunda/service/GroupServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    // when
+    final var searchQueryResult = services.search(searchQuery);
+
+    // then
+    assertThat(searchQueryResult).isEqualTo(result);
+  }
+
+  @Test
+  public void shouldReturnSingleVariable() {
```
Comment (nicpuppa) [inline service/src/test/java/io/camunda/service/GroupServiceTest.java:63]: ❓ I'm not sure what is the scope of the test, can you clarify ?

File: service/src/test/java/io/camunda/service/GroupServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    // given
+    final var entity = mock(GroupEntity.class);
+    final var result = new SearchQueryResult<>(1, List.of(entity), Arrays.array());
+    when(client.searchGroups(any())).thenReturn(result);
+  }
+
+  @Test
+  public void shouldReturnSingleVariableForGet() {
```
Comment (nicpuppa) [inline service/src/test/java/io/camunda/service/GroupServiceTest.java:71]: ❓ I'm not sure what is the scope of the test, can you clarify ?

**Review by nicpuppa — CHANGES_REQUESTED** [review]
Thanks for splitting the PR into separate commits 💪 

Please have a look at my comments 🙏

File: service/src/test/java/io/camunda/service/GroupServiceTest.java (hunk 2 above)
Comment (nicpuppa) [reply service/src/test/java/io/camunda/service/GroupServiceTest.java:28]: Ok I saw you adjusted this in the other PR

File: service/src/test/java/io/camunda/service/GroupServiceTest.java (hunk 3 above)
Comment (nicpuppa) [reply service/src/test/java/io/camunda/service/GroupServiceTest.java:63]: Same as above

File: service/src/test/java/io/camunda/service/GroupServiceTest.java (hunk 4 above)
Comment (nicpuppa) [reply service/src/test/java/io/camunda/service/GroupServiceTest.java:71]: Same as above

**Comment by koevskinikola** [comment]
✅ @nicpuppa all the review feedback is applied.

**Review by nicpuppa — APPROVED** [review]

**Pull request merged**